packages/nimble-components/src/menu-button/testing/menu-button.pageobject.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    keyArrowDown,
+    keyArrowUp,
+    keyEnter,
+    keyEscape,
+    keySpace
+} from '@microsoft/fast-web-utilities';
+import type { MenuButton } from '..';
+import { createEventListener } from '../../utilities/tests/component';
+
+/**
+ * Page object for `nimble-menu-button` component to provide consistent ways
+ * of querying and interacting with the component during tests.
+ *
+ * This page object is intended to be used to interact with a `nimble-menu-button`,
+ * not the menu slotted into the button or the menu items.
+ */
+export class MenuButtonPageObject {
+    public constructor(protected readonly menuButtonElement: MenuButton) {}
+
+    public isOpen(): boolean {
+        return this.menuButtonElement.open;
+    }
+
+    public clickMenuButton(): void {
+        // Focus the menu button before calling click() because ordinarily a mouse click
+        // would bring focus to the button, but calling click() directly does not.
+        this.menuButtonElement.focus();
+        this.menuButtonElement.toggleButton!.control.click();
+    }
+
+    public async openMenu(): Promise<void> {
+        if (this.isOpen()) {
+            return;
+        }
+
+        const toggleListener = createEventListener(
+            this.menuButtonElement,
+            'toggle'
+        );
+        this.clickMenuButton();
+        await toggleListener.promise;
+    }
```

Comment:
I could see clients being unsure which of these they should use if their goal is just to open a menu to click an option. Could we provide some guidance via comments?

I'm guessing that guidance would be to prefer `openMenu()` since it [waits for the `toggle` event which implies the menu is actually visible](https://github.com/ni/nimble/blob/b3cf1500163c84dd7caee3ed350c36afd2979d1b/packages/nimble-components/src/menu-button/index.ts#L145). I can't think of a common use case for clients to want to use the click / keyboard methods so maybe we should discourage their use saying something like "only use this for fine-grained control over event timing".

Reply:
That's a good point. We could also make some of these functions internal if we don't think clients will have a reason to use some of them. I think that's a pattern we've done in other page objects, right?

Reply:
Other thoughts:
- We also did a thing where we made an internal pageObject that we didn't export. We could move some of this to an internal page object
- The Web Crypto api and the Signals api both have a sub reference called `subtle` to put things that aren't expected to be generally used. We could consider that pattern to make those operations still available outside nimble but indicate they aren't expected to be used often.

```ts
class MenuButtonPageObject {
    public constructor(
        protected readonly menuButtonElement: MenuButton
    ) {
        this.subtle = new MenuButtonSubtlePageObject(menuButtonElement);
    }
}
```

Reply:
Yeah, for [select we have a separate internal page object](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/select/tests/select.pageobject.internal.ts) and for other page objects we've just added `@internal` comments. I'm fine with either approach if you can form a sense of which methods would be useful to clients.

Reply:
I'm leaning towards marking some functions as `@internal`. Upon further thought, I think creating a separate page object will likely be convoluted for cases like the menu button column page object where we want to expose the menu button page object as part of the API but also internally call internal functions on the menu button's page object.

Reply:
I suspect that using `subtle` wouldn't be an issue, just have circular references between them. could even keep the classes in the same file, but also don't feel the need to make a new pattern for this PR.

Reply:
I decided to mark quite a few functions as `@internal`.